File: Cmdlets/Cmdlets/CurrencyRegion.vb
Public Class CurrencyRegion

    Private _Currency As TIKSN.Finance.CurrencyInfo

    Private _Region As System.Globalization.RegionInfo

    Public Sub New(Region As System.Globalization.RegionInfo)

        Me._Region = Region
        Me._Currency = New TIKSN.Finance.CurrencyInfo(Region)
    End Sub

    Public ReadOnly Property Currency() As TIKSN.Finance.CurrencyInfo
        Get
            Return Me._Currency
        End Get
    End Property

    Public ReadOnly Property Region() As System.Globalization.RegionInfo
        Get
            Return Me._Region
        End Get
    End Property
End Class

File: Common/Models/CurrencyRegion.cs
namespace TIKSN.Cmdlets.Common.Models
{
    public class CurrencyRegion
    {
        private TIKSN.Finance.CurrencyInfo _Currency;

        private System.Globalization.RegionInfo _Region;

        public CurrencyRegion(System.Globalization.RegionInfo Region)
        {
            this._Region = Region;
            this._Currency = new TIKSN.Finance.CurrencyInfo(Region);
        }

        public TIKSN.Finance.CurrencyInfo Currency
        {
            get { return this._Currency; }
        }

        public System.Globalization.RegionInfo Region
        {
            get { return this._Region; }
        }
    }
}